source/write/bulk-write.txt

```diff
@@ -0,0 +1,287 @@
+         provided. If an error occurs, the remaining operations are not
+         attempted. If ``False``, the driver performs the operations in an
+         arbitrary order and attempts to perform each operation.
+       | Defaults to ``True``.
+
+   * - ``bypass_document_validation``
+     - | Specifies whether the operation bypasses document-level validation. For more
+         information, see :manual:`Schema
+         Validation </core/schema-validation/#schema-validation>` in the MongoDB
+         Server manual.
+       | Defaults to ``False``.
+
+   * - ``session``
+     - | An instance of ``ClientSession``. For more information, see the `API
+         documentation <{+api-root+}pymongo/client_session.html#pymongo.client_session.ClientSession>`__.
+
+   * - ``comment``
+     - | A comment to attach to the operation. For more information, see the :manual:`delete command
+         fields </reference/command/delete/#command-fields>` guide in the
+         MongoDB Server manual for more information.
+
+   * - ``let``
+     - | A map of parameter names and values. Values must be constant or closed
+         expressions that don't reference document fields. For more information,
+         see the :manual:`let statement
+         </reference/command/delete/#std-label-delete-let-syntax>` in the
+         MongoDB Server manual.
+
+The following example uses the ``bulk_write()`` with the ``ordered`` options set
+to ``False``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-write-unordered
+   :end-before: end-bulk-write-unordered
+   :language: python
+   :copyable:
+
+If any of the write operations in the preceding example fail, {+driver-short+}
+reports the errors only after attempting all operations.
+
+.. note::
+
+   Unordered bulk operations do not guarantee order of execution. The order can
+   differ from the way you list them to optimize the runtime.
+
+Return Value
+------------
+
+The ``bulk_write()`` method returns a ``BulkWriteResult`` object. The
+``BulkWriteResult`` object contains the following properties:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``acknowledged``
+     - | Indicates if the server acknowledged the write operation.
+
+   * - ``bulk_api_result``
+     - | The raw bulk API result returned by the server.
+         
+   * - ``deleted_count``
+     - | The number of documents deleted, if any.
+
+   * - ``inserted_count``
+     - | The number of documents inserted, if any.
+
+   * - ``matched_count``
+     - | The number of documents matched for an update, if applicable.
+
+   * - ``modified_count``
+     - | The number of documents modified, if any.
+
+   * - ``upserted_count``
+     - | The number of documents upserted, if any.
+
+   * - ``upserted_ids``
+     - | A map of operation index to the ``_id`` of the upserted documents, if
+         applicable.
```

Comment:
I think there's a syntax error here but I don't know what it is. Maybe "operations indexed"?

Reply:
I took this from the API docs, but I'm pretty sure it's supposed to be a map that maps the index of the operation to the _id of the documents. I updated the wording a bit